docs/interpreter/spark-interpreter-k8s.md

```diff
@@ -0,0 +1,164 @@
+
+ - Docker https://kubernetes.io/docs/tasks/tools/install-minikube/ 
+
+    This documentation uses a pre-built Spark 2.2 Docker images, however you may also build these images as described here: https://github.com/apache-spark-on-k8s/spark/blob/branch-2.2-kubernetes/resource-managers/kubernetes/README.md 
+
+## Checkout Zeppelin source code
+
+Checkout the latest source code from https://github.com/apache/zeppelin then apply changes from the [Add support to run Spark interpreter on a Kubernetes cluster](https://github.com/apache/zeppelin/pull/2637) pull request.
+
+## Build Zeppelin
+- `./dev/change_scala_version.sh 2.11`
+- `mvn clean install -DskipTests -Pspark-2.2 -Phadoop-2.4 -Pyarn -Ppyspark -Pscala-2.11` 
+
+
+## Create distribution
+- `cd zeppelin-distribution`
+- `mvn org.apache.maven.plugins:maven-assembly-plugin:3.0.0:single -P apache-release`
+
+## Create Zeppelin Dockerfile in Zeppelin distribution target folder
+```
+cd {zeppelin_source}/zeppelin-distribution/target/zeppelin-0.8.0-SNAPSHOT
+cat > Dockerfile <<EOF
+FROM banzaicloud/spark-base:v2.2.0-k8s-1.0.179
+COPY zeppelin-0.8.0-SNAPSHOT /opt/zeppelin
+ADD https://storage.googleapis.com/kubernetes-release/release/v1.7.4/bin/linux/amd64/kubectl /usr/local/bin
+WORKDIR /opt/zeppelin
+ENTRYPOINT bin/zeppelin.sh
+EOF
+``` 
+
+## Create / Start a Kubernetes cluster
+In case of using Minikube on Linux with KVM:
+ 
+`minikube start --vm-driver=kvm --cpus={nr_of_cpus} --memory={mem}`
+
+You can check the Kubernetes dashboard address by running: `minikube dashboard`.
+
+Init docker env: `eval $(minikube docker-env)`
+
+## Build & tag Docker image
+
+```
+docker build -t zeppelin-server:v0.8.0-k8s-1.0.34 -f Dockerfile .
+docker tag <imageid> banzaicloud/zeppelin-server:v0.8.0-k8s-1.0.34
+```
+
+You can retrieve the `imageid` by running docker images` 
+
+## Start ResourceStagingServer for spark-submit
+
+Spark-submit will use ResourceStagingServer to distribute resources (in our case the Zeppelin Spark interpreter JAR) across Spark driver and executors.
+
+```
+wget https://github.com/apache-spark-on-k8s/spark/blob/branch-2.2-kubernetes/conf/kubernetes-resource-staging-server.yaml  
+kubectl create -f kubernetes-resource-staging-server.yaml
+``` 
+
+## Create a Kubernetes service to reach Zeppelin server from outside the cluster
+
+``` 
+cat > zeppelin-service.yaml <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: zeppelin-k8-service
+  labels:
+	app: zeppelin-server
+spec:
+  ports:
+  - port: 8080
+	targetPort: 8080
+  selector:
+	app: zeppelin-server
+  type: NodePort
+EOF
+
+kubectl create -f zeppelin-service.yaml
+
+``` 
+
+## Start Zeppelin server
+
+``` 
+cat > zeppelin-pod-local.yaml <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: zeppelin-server
+  labels:
+	app: zeppelin-server
+spec:
+  containers:
+  - name: zeppelin-server
+	image: banzaicloud/zeppelin-server:v0.8.0-k8s-1.0.34
+	env:
+	- name: SPARK_SUBMIT_OPTIONS
```

Comment:
`SPARK_SUBMIT_OPTIONS` is a global setting which affect all the spark interpreter. It would be better to only change the interpreter setting.

Reply:
The assumption here is that you will run all spark interpreters on K8S, I used SPARK_SUBMIT_OPTIONS since this is the default way to add custom params, however, you can also set up these properties in interpreter settings as properties prefixed with 'spark' and will be added as config params automatically.